satpy/readers/seviri_l1b_hrit.py

```diff
@@ -725,18 +794,21 @@ def _get_raw_mda(self):
         return raw_mda
 
     def _get_timestamps(self):
-        """Read scanline timestamps from the segment header"""
+        """Read scanline timestamps from the segment header."""
         tline = self.mda['image_segment_line_quality']['line_mean_acquisition']
         return get_cds_time(days=tline['days'], msecs=tline['milliseconds'])
 
 
-def show(data, negate=False):
-    """Show the stretched data.
-    """
-    from PIL import Image as pil
-    data = np.array((data - data.min()) * 255.0 /
-                    (data.max() - data.min()), np.uint8)
-    if negate:
-        data = 255 - data
-    img = pil.fromarray(data)
-    img.show()
+def pad_hrv_data(data, final_size, east_bound, west_bound):
+    """Pad the data given east and west bounds and the desired size."""
+    nlines = final_size[0]
+    if west_bound - east_bound != data.shape[1] - 1:
+        raise IndexError('East and west bounds do not match data shape')
+    padding_east = da.zeros((nlines, east_bound - 1),
+                            dtype=data.dtype, chunks=CHUNK_SIZE)
+    padding_west = da.zeros((nlines, (final_size[1] - west_bound)),
+                            dtype=data.dtype, chunks=CHUNK_SIZE)
+    if np.issubdtype(data.dtype, np.floating):
+        padding_east = padding_east * np.nan
+        padding_west = padding_west * np.nan
+    return np.hstack((padding_east, data, padding_west))
```

Comment:
Is the numpy hstack required or can this be done using a dask equivalent? Does it even matter? :smile:

Reply:
Fwiu it's recommended to use np variants of the functions when they preserve the dask arrays

Reply:
Ok!